### ISSUE TITLE
Invalid build number for candidate scraper should fail

### ISSUE THREAD

**Issue by whimboo** [mozilla/mozdownload#445]
By running the following code, we incorrectly download `build6` instead of failing because there is no `build2` available.
```
% mozdownload -t candidate -v 53.0 --build-number 2
INFO | Retrieving list of candidate builds from https://archive.mozilla.org/pub/firefox/candidates/53.0-candidates/
INFO | Found 4 builds: build1, build3, build5, build6
INFO | Selected build: build6
INFO | Downloading from: https://archive.mozilla.org/pub/firefox/candidates/53.0-candidates/build6/mac/en-US/Firefox%2053.0.dmg
INFO | Saving as: /Volumes/data/code/mozdownload/firefox-53.0-build6.en-US.mac.dmg
```

**Comment by wardt7** [comment]
Hi,

I've started work on this bug and have forked the repository (the bug does still exist after testing it my end); I was just wondering if there's anything I need to know about the ReleaseCandidateScraper itself? I have an idea on how to solve this by checking to see if ('build%s' % self.build_number) returns anything (and if not throwing an exception), but this seems like a super hacky way to solve the issue and I'm sure there's a better way of doing it.

I'm also happy to create a unit test to make sure this doesn't happen again in the future if you wish.

Toby

**Comment by whimboo** [comment]
Thank you for your interest in working on this issue. You are welcome to get started. Here some additional information regarding your questions... You don't have to do an additional request. Instead check the method which prints out which builds were found (see my first comment too). If the wanted build like `build2` in the above case is not present, it has to raise a not found error, and not download the latest build. A unit test will be greatly welcome too. I hope that helps to clarify it.

**Comment by wardt7** [comment]
Hi,

I've made a pull request for this fix. Is it possible for it to be reviewed, as there are a couple of build failures which I think are errors in the tests themselves rather than issues on my end (although I'm not 100% sure)

Thanks,
Toby

**Comment by whimboo** [comment]
Hi @wardt7. Sorry that I missed your comment here. I will immediately have a look at your PR #513.

**Comment by wardt7** [comment]
Hi @whimboo . Apologies but I'm going to have to give up working on this PR as university and family life are massively getting in the way at the moment. More than happy for people to use my fork as a basis for a patch though, and I'll stay subscribed just in case a new assignee has any questions about my fork or the issue itself.

**Comment by whimboo** [comment]
Thanks for the information! But also sad to hear because it was close to get landed. So thanks for giving a template. I will set this issue as mentored now, so we can get it hopefully fixed soon.

**Comment by andreheringer** [comment]
Does this issue still needs attention? Looks like @Utkarsh1308 commited code that solves it but there is no PR opened.

**Comment by theshteves** [comment]
I'd also be willing to salvage his PR if you need extra hands.

**Comment by whimboo** [comment]
I didn't get any reply from @Utkarsh1308 on PR #541. So feel free to get started on this issue, and please take the comments on that other PR into account. Thanks!

**Comment by sushant23y** [comment]
I am a beginner, how do I get started?

**Comment by whimboo** [comment]
@sushant23y please see the link to the old pull request as given in my last comment. You will have to pick those changes, and apply my review comments on-top of it before creating your own pull request. Thanks!

**Comment by claudius96** [comment]
hie @whimboo i'm a newbie in opensource and would love to get started

**Comment by Divyanshu29** [comment]
Is the issue still same as it was at the starting? Or some changes has been done? Can you just clarify this?

**Comment by Divyanshu29** [comment]
I have applied the patch. Pls suggest if there is any issue!

**Comment by Divyanshu29** [comment]
Hi @whimboo I have made the pull Requests but it seems to fail testcases. Can you help me out?

**Comment by whimboo** [comment]
@Divyanshu29 thanks for the PR. I will have a look at it soon.

@claudius96 if you are still interested in contribution please have a look at other issues on this repository. I'm happy to mentor you on any of those. Thanks.

**Comment by Divya063** [comment]
@whimboo  I would like to work on this issue.

**Comment by whimboo** [comment]
@Divya063 sounds good. Let me know if you have questions. Please take a look at https://github.com/mozilla/mozdownload/pull/564 in how to pick-up the work. Thanks.

**Comment by Divya063** [comment]
@whimboo Sure thanks!

**Comment by Xk0nSid** [comment]
@whimboo Does this issue still requires attention because I would like to work on this.

**Comment by whimboo** [comment]
Sorry that I forget to reply. Yes, right now no-one is working on this issue. So feel free to pick it up. Thanks!

**Comment by whimboo** [comment]
Fixed by the merge of PR #607.
